### PR TITLE
Sync sanity scenarios

### PR DESCRIPTION
**Pull request by sethrosetter**
- Add sanity test for access() in sync function
- Add sanity test for channel() in sync function
- Add sanity test for role() in sync function
- Add requireRoles() sanity test
- Add ignore_rev_ids=False to verify changes. This is to ignore rev_id checking for the situation where the user verify their changes includes doc that they did not create or update
- Add verify_docs_removed function (Needs REVIEW) to check a users changes feed that had documents in channels that were changed

### CODE REVIEW

**Comment by sethrosetter** [comment]
@ashvindersingh this has been rebased off of master

File: lib/verify.py [hunk 1]
```diff
@@ -22,7 +22,106 @@ def verify_same_docs(expected_num_docs, doc_dict_one, doc_dict_two):
...
+
+        # Get ids from expected docs
+        expected_doc_ids = expected_docs.keys()
+
+        # Assert there are no duplicates in expected doc ids
+        if len(expected_doc_ids) != len(set(expected_doc_ids)):
+            errors["duplicate_expected_ids"] += 1
+
+        # Get ids from all changes results
+        changes_doc_ids = [result["id"] for result in changes_results]
```
Comment (ashvindersingh) [inline lib/verify.py]: this can be done when parsing the changes feed results.

File: lib/verify.py [hunk 2]
```diff
@@ -22,7 +22,106 @@ def verify_same_docs(expected_num_docs, doc_dict_one, doc_dict_two):
...
+            elif result["removed"] is True:
+                num_doc_removed += 1
+
+            # Compare revision number for id
+            if expected_docs[result["id"]] != result["rev"]:
+                errors["invalid_rev_id"] += 1
+
+            # TODO - maybe try to ping doc endpoint and asser 4XX response?
+
+        print(" -> REMOVED |{0}| expected (num_docs: {1}) _changes (num_docs: {2}, num_removed: {3})".format(
```
Comment (ashvindersingh) [inline lib/verify.py]: we should use log function instead of print.

File: lib/verify.py [hunk 3]
```diff
@@ -22,7 +22,106 @@ def verify_same_docs(expected_num_docs, doc_dict_one, doc_dict_two):
     print(" -> doc_dict_one == doc_dict_two expected (num_docs: {})".format(expected_num_docs))
 
 
-def verify_changes(users, expected_num_docs, expected_num_revisions, expected_docs):
+def verify_docs_removed(users, expected_num_docs, expected_docs):
```
Comment (ashvindersingh) [inline lib/verify.py]: this function should verify that tombstone revision doesn't change after the doc has been marked deleted.

File: lib/verify.py [hunk 4]
```diff
@@ -90,10 +189,15 @@ def verify_changes(users, expected_num_docs, expected_num_revisions, expected_do
         if set(expected_doc_ids) != set(changes_doc_ids):
             errors["expected_doc_ids_differ_from_changes_doc_ids"] += 1
 
+        if ignore_rev_ids:
+            print("WARNING: Ignoring rev id verification!!")
```
Comment (ashvindersingh) [inline lib/verify.py]: same as above comment, use log instead of print.

File: lib/verify.py [hunk 5]
```diff
@@ -48,6 +151,7 @@ def verify_changes(users, expected_num_docs, expected_num_revisions, expected_do
         user_list.append(users)
 
     if type(expected_docs) is not dict:
+        log.error("expected_docs is not a dictionary")
```
Comment (ashvindersingh) [inline lib/verify.py]: It would be helpful to also log the user-id

**Pull request merged**